frontend/simple/views/containers/contributions/IncomeForm.vue

```diff
@@ -0,0 +1,246 @@
+
+  @include mobile {
+    height: 100vh;
+  }
+
+  &-columns {
+    @include mobile {
+      flex-direction: column;
+    }
+  }
+}
+
+.c-help-extra {
+  margin-top: $gi-spacer-sm;
+}
+
+.c-form,
+.c-pledgesGraph {
+  flex-basis: 50%;
+}
+
+.c-pledgesGraph {
+  align-items: flex-start;
+  flex-grow: 1;
+
+  @include mobile {
+    order: -1; // display on the top for better understanding on small screens
+    padding: $gi-spacer $gi-spacer-sm;
+  }
+
+  @include tablet {
+    justify-content: center;
+    margin-top: -$gi-spacer-lg;
+  }
+}
+
+</style>
+<script>
+import { validationMixin } from 'vuelidate'
+import { requiredIf, required } from 'vuelidate/lib/validators'
+import InputAmount from './InputAmount.vue'
+import PaymentMethods from './PaymentMethods.vue'
+import Modal from '../../components/Modal/ModalBasic.vue'
+import TextWho from '../../components/TextWho.vue'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+const fakeStoreMincome = 500
+
+const hasLowIncome = (value) => value < fakeStoreMincome - 1
+
+export default {
+  name: 'IncomeForm',
+  mixins: [ validationMixin ],
+  components: {
+    Modal,
+    InputAmount,
+    TextWho,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  props: {
+    isEditing: Boolean
+  },
+  data () {
+    return {
+      ephemeral: {
+        /* formVerified: false */
+      },
+      form: {
+        option: null,
+        income: null,
+        pledge: null
+      },
+      // -- Hardcoded Data just for layout purposes:
+      fakeStore: {
+        mincome: fakeStoreMincome,
+        mincomeFormatted: '$500'
+      }
+    }
+  },
+  computed: {
+    infoRequired () {
+      return this.L('This information is required.')
+    },
+    needsIncome () {
+      return this.form.option === 'no'
+    },
+    canPledge () {
+      return this.form.option === 'yes'
+    },
+    inputIncomeError () {
+      if (!this.$v.form.income.hasLowIncome) {
+        return this.L('If your income is the same or higher than the group\'s mincome change your answer to "Yes, I do"')
+      } else if (this.$v.form.income.$error) {
+        return this.infoRequired
+      }
+    },
+    saveButtonText () {
+      const verb = this.isFirstTime ? 'Add' : 'Save'
+
+      if (!this.form.option) { return this.L('{verb} details', { verb }) }
+      if (this.canPledge) { return this.L('{verb} pledged details', { verb }) }
+      if (this.needsIncome) { return this.L('{verb} income details', { verb }) }
+    }
+  },
+  methods: {
+    verifyInputIncome (e) {
+      this.form.income = e.target.value
+      this.$v.form.income.$touch()
+    },
+    verifyInputPledge (e) {
+      this.form.pledge = e.target.value
+      this.$v.form.pledge.$touch()
+    },
+    verifyForm () {
+      this.$v.form.$touch()
+
+      if (!this.$v.form.$invalid) {
+        const canPledge = this.canPledge
+
+        this.$emit('save', {
+          canPledge,
+          amount: canPledge ? this.$v.form.pledge.$model : this.$v.form.income.$model
```

Comment:
Again, within the javascript section of the component (this whole section), this style of accessing data (e.g. `this.$v.form.pledge.$model`) should not be done.

It should be accessed directly instead, like so: `this.form.pledge`.

Reply:
Replaced those and everywhere with `.$model`